packages/zoe/test/unitTests/contracts/escrowToVote.js

```diff
@@ -0,0 +1,118 @@
+/* global harden */
+// @ts-check
+
+// Eventually will be importable from '@agoric/zoe-contract-support'
+import { assert } from '@agoric/assert';
+import makeStore from '@agoric/store';
+import { makeZoeHelpers } from '../../../src/contractSupport';
+
+/**
+ * Implement coin voting. Give a voting capability when a payment is
+ * escrowed.
+ *
+ * @typedef {import('../../../src/zoe').ContractFacet} ContractFacet
+ * @typedef {import('@agoric/ERTP').Amount} Amount
+ * @param {ContractFacet} zcf
+ */
+const makeContract = zcf => {
+  const { assertKeywords, checkHook } = makeZoeHelpers(zcf);
+  assertKeywords(harden(['Assets']));
+  const {
+    terms: { question },
+  } = zcf.getInstanceRecord();
+
+  const offerHandleToResponse = makeStore('offerHandle');
+
+  // `question` is a string defined in the terms.
+  // We assume the only valid answers are 'YES' and 'NO'
+  const validateResponse = response => {
+    assert(
+      Object.keys(response).length === 1 &&
+        Object.keys(response)[0] === question,
+      `The question '${
```

Comment:
This is a rather perfect occasion to use `details` with `q`. Since the question may be an arbitrary string, possibly including quote characters, the question itself should appear quoted with proper escaping of embedded quotes. Since there's no reason to hide the question from those who might see the thrown error, `details` + `q` does exactly this.

In general, for the complaint in this position of an assert, if it isn't a `details` template literal, I prefer it not to be a plain template literal unless there's a reason. If someone reading the code misses the absence of the expected leading `details` they may assume substitution values are hidden when they're not.

Both comments apply to several other cases below.

Reply:
Ah, right! Thanks, that is good to know. Could we have some more documentation on q on when to use it and what it is? I have an issue for that here, but didn't know who to assign it to. It might be helpful to get a first write from an engineer that Tom can later fix. https://github.com/Agoric/documentation/issues/230